src/containers/SettingsPage/index.js

```diff
@@ -1,5 +1,43 @@
 import { connect } from 'react-redux'
+import { reduxForm, formValueSelector, submit } from 'redux-form'
 
 import Settings from 'components/Settings'
+import { updateSettings } from 'actions/settings'
 
-export default connect()(Settings)
+const FORM = {
+  form: 'settingsForm',
+  onChange: async (values, dispatch, props) => {
+    const { formValues, syncErrors, pristine, dirty } = props
+    // TODO should only if no errors but seems to be an issue with
+    // onchange validation
+    // https://github.com/erikras/redux-form/issues/3012
+    return await dispatch(updateSettings(formValues.settings))
```

Comment:
Instead of validating onChange you can instead validate on "onSubmit" and change this handler to immediately submit the form, then you can use the validator function here.

(I think `.submit()` exists in props, otherwise import the action creator to submit a form.)